lib/rules/jsx-no-literals.js

```diff
@@ -40,8 +40,14 @@ module.exports = {
   },
 
   create(context) {
+    function trimIfString(val) {
+      return typeof val === 'string' ? val.trim() : val;
+    }
+
     const isNoStrings = context.options[0] ? context.options[0].noStrings : false;
-    const allowedStrings = context.options[0] ? new Set(context.options[0].allowedStrings) : false;
+    const allowedStrings = context.options[0] ?
+      new Set((context.options[0].allowedStrings || []).map(trimIfString)) :
```

Comment:
why the ternary? just making this unconditionally be a Set simplifies the check below